feat: Use `keyring` module from `@zowe/secrets-for-zowe-sdk` to replace use of `node-keytar`

traeok opened the pull request:
## Proposed changes

This PR:
- implements the use of the `keyring` module from the new [`@zowe/secrets-for-zowe-sdk`](https://github.com/zowe/zowe-cli/tree/master/packages/secrets)
- removes old references to `node-keytar` in the codebase and tests
- updates old `node-keytar` references to the new `keyring` module

**Note** that tests will fail until the imperative PR (linked below) is merged and published.

## Release Notes

Milestone: 2.10.0

Changelog: TBD

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [x] New feature (non-breaking change which adds functionality)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [x] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [x] There is coverage for the code that I have added
- [x] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [x] Any PR dependencies have been merged and published (if appropriate)

zFernand0 reviewed (commented):
Changes LGTM! 😋 
will test later today 🥳

adam-wolfe commented:
Checked this out with VS Code Insiders and the --disable-keytar option. LGTM.

zFernand0 commented:
for those on a headless environment, you might want to take a look at:
- https://docs.zowe.org/v2.7.x/user-guide/cli-configure-scs-on-headless-linux-os/#unlocking-the-keyring-automatically
- https://wiki.archlinux.org/title/GNOME/Keyring#Automatically_change_keyring_password_with_user_password

zFernand0 reviewed (dismissed):
Assuming [these changes](https://github.com/zowe/vscode-extension-for-zowe/compare/feat/keytar-rs...env-suggestion?expand=1), I was able to share credentials between the CLI and ZE in my remote SSH (headless) environment 🥳 

LGTM! 😋

traeok commented:
> Assuming [these changes](https://github.com/zowe/vscode-extension-for-zowe/compare/feat/keytar-rs...env-suggestion?expand=1), I was able to share credentials between the CLI and ZE in my remote SSH (headless) environment 🥳
> 
> LGTM! 😋

Thanks for looking into the D-bus env. variable @zFernand0! Do you mind if I cherry-pick this commit into this branch? I'll leave you as the commit author, of course 😋

zFernand0 commented:
> > Assuming [these changes](https://github.com/zowe/vscode-extension-for-zowe/compare/feat/keytar-rs...env-suggestion?expand=1), I was able to share credentials between the CLI and ZE in my remote SSH (headless) environment 🥳
> > LGTM! 😋
> 
> Thanks for testing & looking into the D-bus env. variable zFernand0! Do you mind if I cherry-pick [bb9f963](https://github.com/zowe/vscode-extension-for-zowe/commit/bb9f963c64c76febe3731d81e54a64cf19db268c) into this branch? I'll leave you as the commit author, of course 😋

Yeah, feel free. 😋 
Thanks for considering it! 🙏🏽

t1m0thyj commented on packages/zowe-explorer/package.json:
```diff
@@ -1905,12 +1905,12 @@
   },
   "scripts": {
-    "build": "gulp build && yarn license && webpack --mode development",
+    "build": "gulp build && yarn license && ts-node ./scripts/getSecretsPrebuilds.ts && webpack --mode development",
```
Do we want to try [copy-webpack-plugin](https://webpack.js.org/plugins/copy-webpack-plugin/) instead of the custom script?

t1m0thyj reviewed (approved):
LGTM, thanks @traeok!

<img width="440" alt="image" src="https://github.com/zowe/vscode-extension-for-zowe/assets/22344007/f8e1c65a-655b-4fc1-84f3-7a8047bcfb56">

The "Electron Helper (Plugin)" was unexpected 😄 Maybe it happened because I was running the extension in VS Code debugger? Regardless I don't think it matters because "Zowe" is in the message so it's clear what it is trying to access 👍

traeok commented:
I've updated this PR to address the `copy-webpack-plugin` comment. Since I've also updated the PR to use `7.18.2` of the Secrets SDK, the stages will fail until 7.18.2 is released.
